contracts/src/Payment.sol

```diff
@@ -159,4 +170,54 @@ contract Payment {
         Channel storage channel = channels[publisher][provider][podId];
         return channel.withdrawnByProvider;
     }
+
+    // authorize other addresses to create subChannels
+    function authorize(address _authorized, address provider, bytes32 podId) public {
+        Channel storage channel = channels[msg.sender][provider][podId];
+        if (channel.investedByPublisher == 0) revert DoesNotExist();
+        channel.authorized[_authorized] = true;
+    }
+
+    function isAuthorized(address publisher, address provider, bytes32 podId, address _address)
+        public
+        view
+        returns (bool)
+    {
+        return channels[publisher][provider][podId].authorized[_address];
+    }
+
+    // create a subChannel from a main channel
+    function createSubChannel(
+        address publisher,
+        address provider,
+        bytes32 podId,
+        address newProvider,
+        bytes32 newPodId,
+        uint256 amount
+    ) public {
+        Channel storage channel = channels[publisher][provider][podId];
+        // Ensure the channel exists
+        if (channel.investedByPublisher == 0) revert DoesNotExist();
+
+        // Check if the caller is authorized
+        if (!channel.authorized[msg.sender]) revert NotAuthorized();
+
+        // Ensure there is enough invested by the publisher
+        if (channel.investedByPublisher < amount) revert InsufficientFunds();
+
+        // Deduct the amount from the main channel's funds
+        channel.investedByPublisher -= amount;
+
+        // Create the subChannel for the authorized caller
+        Channel storage subChannel = channels[msg.sender][newProvider][newPodId];
+
+        // authorize the main channel publisher to control the subchannel
+        subChannel.authorized[publisher] = true;
+
+        // fund the new sub channel with the deducted amount from the main channel
+        subChannel.investedByPublisher += amount;
+        subChannel.unlockTime = channel.unlockTime; // Inherit unlock time from main channel
```

Comment:
If I'm reading the code right, this lets the publisher instantly unlock any channel, by doing (in all below calls, `msg.sender == publisher`):
0. `createChannel(provider, podId, ...)` (created earlier)
1. `createChannel(publisher, 0x0, 0, 1)` (create a dummy channel with unlock time 0)
2. `authorize(publisher, publisher, 0x0)` (authorize someone for it (in this case, ourselves))
3. `createSubChannel(publisher, publisher, 0x0, provider, podId)` (oops! resets unlockTime on the original channel to 0)
4. `unlock(publisher, provider, podId)` (regular unlock)
5. `withdrawUnlocked(publisher, provider, podId)` (oops! can withdraw supposedly-locked funds without waiting for unlockTime) (would work even if made in the same block as all the other commands, with no chance for the provider to cash out any work done since the last `withdraw(..)` call)

Reply:
Correct, Should be Fixed now.